### DISCUSSION TITLE
Upload file functionality in UI

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#519] [Ideas]
Hi Team & @drakkan 

The UI looks promising and thanks for that. But can we get an upload file functionality. It would be great to have.

Thanks,
Rashmi Maharanab

**Comment by drakkan** [comment]
Hi,

the development version already has this feature, please give a try to `edge` docker image variant or download a recent build artifact and let me know if it works as expected, thank you

**Reply by rashmimaharana** [reply]
Thanks @drakkan . Its working fine. 
Does it has the same fix for below one which we discussed before. Virtual folder sharing in GCS. Just wanted confirm.
https://github.com/drakkan/sftpgo/issues/510

**Reply by drakkan** [reply]
I confirm

**Reply by rashmimaharana** [reply]
Thanks @drakkan for your response. Few more questions.

1> How can we disable some of the functionality from client screen like folder creation or fileupload thru configuration.
2> What's the difference between edge docker image and released one **sftpgo_v2.1.0_linux_arm64.tar.xz** wrt file upload functionality. Because we got that functionality in docker image but not in the released version.
3> Is the client UI secured enough, if not how can we make it more secure. As we don't want to breach security while giving client UI to end user/customer.

**Reply by drakkan** [reply]
> Thanks @drakkan for your response. Few more questions.
> 
> 1> How can we disable some of the functionality from client screen like folder creation or fileupload thru configuration.

you can remove the `upload` and `create_dirs` permissions

> 2> What's the difference between edge docker image and released one **sftpgo_v2.1.0_linux_arm64.tar.xz** wrt file upload functionality. Because we got that functionality in docker image but not in the released version.

edge variant is the development version, you can download build artifacts for the development version from [here](https://github.com/drakkan/sftpgo/actions/runs/1176933620) or build yourself

> 3> Is the client UI secured enough, if not how can we make it more secure. As we don't want to breach security while giving client UI to end user/customer.

I suggest to provide a certificate and enable https, this [howto](https://github.com/drakkan/sftpgo/blob/main/docs/howto/lets-encrypt-certificate.md) could help. The web ui should not be affected from slow loris style attacks, you could configure a reverse proxy using nginx or something similar if you are paranoid

**Comment by sselva** [comment]
@drakkan, love the upload UI. Roughly, when can we expect a release version with that feature?

**Reply by drakkan** [reply]
> @drakkan, love the upload UI. Roughly, when can we expect a release version with that feature?

not soon sorry, there are many other features to be implemented before releasing v2.2
